docs/router/plugins.mdx

```diff
@@ -0,0 +1,243 @@
+
+These plugins implement GraphQL resolvers that can access any data source or service, all while communicating with the router through a high-performance gRPC interface.
+
+The plugin architecture is built on HashiCorp's battle-tested [go-plugin framework](https://github.com/hashicorp/go-plugin), which powers production systems like Vault and Terraform with millions of deployments worldwide. This provides enterprise-grade reliability, secure process isolation, and proven stability for your GraphQL infrastructure.
+
+## Motivation
+
+The biggest pain point in adopting GraphQL isn't GraphQL itself it's the steep learning curve and fragmented subgraph tooling across languages and platforms.
+
+<CardGroup cols={1}>
+  <Card title="Simplify GraphQL Implementation" icon="wand-magic-sparkles">
+    While everyone understands the value of "one schema, one query," building and maintaining production-grade subgraphs across diverse environments remains challenging. Spec support, runtime performance, and type safety depend on your subgraph framework quality.
+  </Card>
+
+  <Card title="Design with GraphQL, Implement with gRPC" icon="code">
+    Our approach combines GraphQL's schema-first flexibility with gRPC's performance and reliability. Using the same plugin system that powers HashiCorp Vault and Terraform, you can develop subgraphs in any language and run them directly in the Cosmo Router.
+  </Card>
+
+  <Card title="Proto-Based Code Generation" icon="bolt">
+    The strict typing and automatic code generation from Protocol Buffers (proto) definitions provides an immense productivity boost, especially in the era of generative AI. The generated proto-based gRPC code creates a strongly-typed foundation that AI tools can more effectively understand, extend, and modify, reducing development time and errors while ensuring consistent API implementation.
+  </Card>
+</CardGroup>
+
+### Benefits Over Traditional Subgraphs
+
+When compared to traditional standalone subgraphs, gRPC plugins offer substantial technical advantages:
+
+<CardGroup cols={2}>
+  <Card title="Simplified Architecture" icon="cloud-arrow-up">
+    Maintain fewer components with unified deployment and monitoring. You can run multiple plugins by the same router
+  </Card>
+  <Card title="Performance" icon="rocket">
+    Achieve significantly lower latency with direct RPC-based communication. The GraphQL framework overhead is eliminated.
+  </Card>
+  <Card title="Enhanced Developer Experience" icon="code">
+    Work with consistent tooling and deployment patterns across all subgraphs. If your team knows how to work with gRPC, they know how to build subgraphs.
+  </Card>
+  <Card title="Seamless Integration" icon="bridge">
+    Connect to legacy systems, databases, and APIs without intermediate GraphQL services. The plugin can be written in any language that supports gRPC.
```

Comment:
This is conflicting with below "**Language Support**: Currently, only Go is supported for plugin development"